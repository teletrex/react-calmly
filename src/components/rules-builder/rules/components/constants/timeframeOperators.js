/* <LICENSE>
* 
* Copyright (C) 2022 Louis F. Roehrs, All rights reserved.
* 
* </LICENSE>
*  */




import { RULE_INPUT_TYPES } from '../../../constants';

const { NONE, RANGE, SINGLE_VALUE, WHEN_RELATION, WITHIN_PERIOD } = RULE_INPUT_TYPES;

export const TIMEFRAME_OPERATORS = [
  // Within:
  {
    inputType: WITHIN_PERIOD,
    label: 'within the last',
    value: 'WITHIN_THE_LAST',
  },

  // Today:
  {
    inputType: NONE,
    label: 'on today',
    value: 'ON_TODAY',
  },
  {
    inputType: NONE,
    label: 'before today',
    value: 'BEFORE_TODAY',
  },

  // Any time:
  {
    inputType: NONE,
    label: 'at any time',
    value: 'AT_ANY_TIME',
  },

  // Date:
  {
    inputType: SINGLE_VALUE,
    label: 'on',
    value: 'ON',
  },
  {
    inputType: SINGLE_VALUE,
    label: 'after',
    value: 'AFTER',
  },
  {
    inputType: SINGLE_VALUE,
    label: 'before',
    value: 'BEFORE',
  },

  // Date range:
  {
    inputType: RANGE,
    label: 'between',
    value: 'BETWEEN',
  },

  // When relation:
  {
    inputType: WHEN_RELATION,
    label: 'exactly',
    value: 'EXACTLY',
  },
  {
    inputType: WHEN_RELATION,
    label: 'more than',
    value: 'MORE_THAN',
  },
  {
    inputType: WHEN_RELATION,
    label: 'less than',
    value: 'LESS_THAN',
  },
];
