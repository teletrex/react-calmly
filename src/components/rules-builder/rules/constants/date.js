/* <LICENSE>
* 
* Copyright (C) 2022 Louis F. Roehrs, All rights reserved.
* 
* </LICENSE>
*  */




export const DATE_PATTERN = 'd{1,2}/d{1,2}/d{4}';
