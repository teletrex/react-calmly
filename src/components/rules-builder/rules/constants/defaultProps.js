/* <LICENSE>
* 
* Copyright (C) 2022 Louis F. Roehrs, All rights reserved.
* 
* </LICENSE>
*  */




export const DEFAULT_PROPS = {
  INPUT_FIELD_VALIDATION: [],
  OPERATORS: [],
  TIMEFRAME: {},
};
