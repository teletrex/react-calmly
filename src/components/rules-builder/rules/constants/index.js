/* <LICENSE>
* 
* Copyright (C) 2022 Louis F. Roehrs, All rights reserved.
* 
* </LICENSE>
*  */




export * from './date';
export * from './defaultProps';
export * from './periods';
export * from './propTypes';
export * from './ruleOperators';
export * from './validationMessages';
export * from './values';
export * from './whenRelations';
