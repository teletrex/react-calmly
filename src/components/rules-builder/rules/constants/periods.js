/* <LICENSE>
* 
* Copyright (C) 2022 Louis F. Roehrs, All rights reserved.
* 
* </LICENSE>
*  */




export const PERIODS = [
  {
    label: 'days',
    value: 'DAYS',
  },
  {
    label: 'weeks',
    value: 'WEEKS',
  },
  {
    label: 'months',
    value: 'MONTHS',
  },
];
