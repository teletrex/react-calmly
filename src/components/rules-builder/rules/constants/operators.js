/* <LICENSE>
* 
* Copyright (C) 2022 Louis F. Roehrs, All rights reserved.
* 
* </LICENSE>
*  */




import { RULE_INPUT_TYPES } from '../../constants';

const {
  MULTIPLE_VALUES,
  NONE,
  PERIOD_NUMBER,
  PERIOD_RANGE,
  RANGE,
  SINGLE_VALUE,
  WHEN_RELATION,
  WITHIN_PERIOD,
} = RULE_INPUT_TYPES;

// Common operators
export const BLANK_OPERATORS = [
  {
    inputType: NONE,
    label: 'is blank',
    value: 'IS_BLANK',
  },
  {
    inputType: NONE,
    label: 'is not blank',
    value: 'IS_NOT_BLANK',
  },
];

export const EXACT_OPERATORS = [
  {
    inputType: WHEN_RELATION,
    label: 'is exactly',
    value: 'IS_EXACTLY',
  },
  {
    inputType: WHEN_RELATION,
    label: 'is not exactly',
    value: 'IS_NOT_EXACTLY',
  },
];

export const IS_OPERATORS = [
  {
    inputType: SINGLE_VALUE,
    label: 'is',
    value: 'IS',
  },
  {
    inputType: SINGLE_VALUE,
    label: 'is not',
    value: 'IS_NOT',
  },
];

export const HAS_OPERATORS = [
  {
    inputType: SINGLE_VALUE,
    label: 'has',
    value: 'HAS',
  },
  {
    inputType: SINGLE_VALUE,
    label: 'has not',
    value: 'HAS_NOT',
  },
];

export const IS_ONE_OF_OPERATORS = [
  {
    inputType: MULTIPLE_VALUES,
    label: 'is one of the following',
    value: 'IS_ONE_OF_THE_FOLLOWING',
  },
  {
    inputType: MULTIPLE_VALUES,
    label: 'is not one of the following',
    value: 'IS_NOT_ONE_OF_THE_FOLLOWING',
  },
];

export const CONTAIN_OPERATORS = [
  {
    inputType: SINGLE_VALUE,
    label: 'contains',
    value: 'CONTAINS',
  },
  {
    inputType: SINGLE_VALUE,
    label: 'does not contain',
    value: 'DOES_NOT_CONTAIN',
  },
];

export const CONTAIN_ONE_OF_OPERATORS = [
  {
    inputType: MULTIPLE_VALUES,
    label: 'contains one of the following',
    value: 'CONTAINS_ONE_OF_THE_FOLLOWING',
  },
  {
    inputType: MULTIPLE_VALUES,
    label: 'does not contain one of the following',
    value: 'DOES_NOT_CONTAIN_ONE_OF_THE_FOLLOWING',
  },
];

export const RANGE_OPERATORS = [
  {
    inputType: RANGE,
    label: 'is between',
    value: 'IS_BETWEEN',
  },
];

// String related operators
export const STARTS_WITH_OPERATORS = [
  {
    inputType: SINGLE_VALUE,
    label: 'starts with',
    value: 'STARTS_WITH',
  },
  {
    inputType: SINGLE_VALUE,
    label: 'does not start with',
    value: 'DOES_NOT_START_WITH',
  },
];

export const STARTS_WITH_ONE_OF_OPERATORS = [
  {
    inputType: MULTIPLE_VALUES,
    label: 'starts with one of the following',
    value: 'STARTS_WITH_ONE_OF_THE_FOLLOWING',
  },
  {
    inputType: MULTIPLE_VALUES,
    label: 'does not start with one of the following',
    value: 'DOES_NOT_START_WITH_ONE_OF_THE_FOLLOWING',
  },
];

export const ENDS_WITH_OPERATORS = [
  {
    inputType: SINGLE_VALUE,
    label: 'ends with',
    value: 'ENDS_WITH',
  },
  {
    inputType: SINGLE_VALUE,
    label: 'does not end with',
    value: 'DOES_NOT_END_WITH',
  },
];

export const ENDS_WITH_ONE_OF_OPERATORS = [
  {
    inputType: MULTIPLE_VALUES,
    label: 'ends with one of the following',
    value: 'ENDS_WITH_ONE_OF_THE_FOLLOWING',
  },
  {
    inputType: MULTIPLE_VALUES,
    label: 'does not end with one of the following',
    value: 'DOES_NOT_END_WITH_ONE_OF_THE_FOLLOWING',
  },
];

// Number related operators
export const LESS_THAN_OPERATORS = [
  {
    inputType: SINGLE_VALUE,
    label: 'is less than',
    value: 'IS_LESS',
  },
  {
    inputType: SINGLE_VALUE,
    label: 'is less than or equal to',
    value: 'IS_LESS_OR_EQUAL',
  },
];

export const GREATER_THAN_OPERATORS = [
  {
    inputType: SINGLE_VALUE,
    label: 'is greater than',
    value: 'IS_GREATER',
  },
  {
    inputType: SINGLE_VALUE,
    label: 'is greater than or equal to',
    value: 'IS_GREATER_OR_EQUAL',
  },
];

// Date related operators
export const TODAY_OPERATORS = [
  {
    inputType: NONE,
    label: 'is today',
    value: 'IS_TODAY',
  },
  {
    inputType: NONE,
    label: 'is after today',
    value: 'IS_AFTER_TODAY',
  },
  {
    inputType: NONE,
    label: 'is not after today',
    value: 'IS_NOT_AFTER_TODAY',
  },
  {
    inputType: NONE,
    label: 'is before today',
    value: 'IS_BEFORE_TODAY',
  },
  {
    inputType: NONE,
    label: 'is not before today',
    value: 'IS_NOT_BEFORE_TODAY',
  },
  {
    inputType: NONE,
    label: 'anniversary is today',
    value: 'ANNIVERSARY_IS_TODAY',
  },
];

export const ANNIVERSARY_OPERATORS = [
  {
    inputType: WHEN_RELATION,
    label: 'anniversary is',
    value: 'ANNIVERSARY_IS',
  },
];

export const WITHIN_OPERATORS = [
  {
    inputType: WITHIN_PERIOD,
    label: 'is within the last',
    value: 'IS_WITHIN_THE_LAST',
  },
  {
    inputType: WITHIN_PERIOD,
    label: 'is not within the last',
    value: 'IS_NOT_WITHIN_THE_LAST',
  },
  {
    inputType: WITHIN_PERIOD,
    label: 'is within the next',
    value: 'IS_WITHIN_THE_NEXT',
  },
  {
    inputType: WITHIN_PERIOD,
    label: 'is not within the next',
    value: 'IS_NOT_WITHIN_THE_NEXT',
  },
];

export const WITHIN_RANGE_OPERATORS = [
  {
    inputType: PERIOD_RANGE,
    label: 'is within the range of the last',
    value: 'IS_WITHIN_THE_RANGE_OF_THE_LAST',
  },
];

export const PERIOD_EQUAL_OPERATORS = [
  {
    inputType: PERIOD_NUMBER,
    label: 'year is',
    value: 'YEAR_IS',
  },
  {
    inputType: PERIOD_NUMBER,
    label: 'month is',
    value: 'MONTH_IS',
  },
  {
    inputType: PERIOD_NUMBER,
    label: 'day is',
    value: 'DAY_IS',
  },
];

export const BEFORE_AFTER_DATE_OPERATORS = [
  {
    inputType: SINGLE_VALUE,
    label: 'is after',
    value: 'IS_AFTER',
  },
  {
    inputType: SINGLE_VALUE,
    label: 'is before',
    value: 'IS_BEFORE',
  }
]

export const DATA_AGGREGATION_OPERATORS = [
  {
    inputType: SINGLE_VALUE,
    label: 'by',
    value: 'BY',
  },

];
