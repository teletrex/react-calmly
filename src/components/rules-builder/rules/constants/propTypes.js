/* <LICENSE>
* 
* Copyright (C) 2022 Louis F. Roehrs, All rights reserved.
* 
* </LICENSE>
*  */




import PropTypes from 'prop-types';

import { RULE_INPUT_TYPES } from '../../constants';

const { any, arrayOf, func, oneOf, oneOfType, shape, string } = PropTypes;

const OPERATOR = oneOfType([
  string,
  shape({
    inputType: oneOf(Object.values(RULE_INPUT_TYPES)).isRequired,
    label: string.isRequired,
    value: string.isRequired,
  }),
]);
const OPERATORS = arrayOf(OPERATOR);

const TIMEFRAME = shape({
  data: shape({
    operator: string,
    value: any,
  }),
  onChange: func,
});

const OPERATOR_NAME = string;

const SELECT_OPTION = shape({
  label: string.isRequired,
  value: string.isRequired,
});

const SELECT_OPTIONS = arrayOf(SELECT_OPTION);

export const PROPTYPES = {
  OPERATOR,
  OPERATORS,
  OPERATOR_NAME,
  SELECT_OPTION,
  SELECT_OPTIONS,
  TIMEFRAME,
};
