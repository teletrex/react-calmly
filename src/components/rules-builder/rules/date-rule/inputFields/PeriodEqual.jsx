/* <LICENSE>
* 
* Copyright (C) 2022 Louis F. Roehrs, All rights reserved.
* 
* </LICENSE>
*  */




import React, { useContext, useEffect } from 'react';
import PropTypes from 'prop-types';
import isEqual from 'lodash/isEqual';
import * as Yup from 'yup';

// import NumberInput from '../../../../number-input';
import { NumberInput } from '@carbon/react';
import { useTranslation} from  '../../../../../translation';
import RulesBuilderContext from '../../../RulesBuilderContext';
import { PROPTYPES, VALIDATION_MESSAGES } from '../../constants';
import { getPrefixedClasses, useFormValidation } from '../../helpers';

const defaultInputProps = {
  DAY_IS: {
    min: 1,
    max: 31,
  },
  MONTH_IS: {
    min: 1,
    max: 12,
  },
  YEAR_IS: {
    min: 1900,
    max: 3000,
  },
};

const getValueSchema = periodValue => {
  const { max, min } = defaultInputProps[periodValue];
  return Yup.number()
    .min(min, VALIDATION_MESSAGES.min)
    .max(max, VALIDATION_MESSAGES.max);
};

const PeriodEqual = ({ id, onValueChange, operator, value }) => {
  const { t } = useTranslation();
  const { isReadOnly } = useContext(RulesBuilderContext);

  const validationSchema = Yup.object().shape({
    [id]: getValueSchema(operator.value),
  });

  const {
    errors,
    handleBlur,
    initialValues,
    resetForm,
    setFieldValue,
    submitForm,
    values,
  } = useFormValidation({
    initialValues: {
      [id]: value,
    },
    validationSchema,
    onSubmit: values => {
      if (!isEqual(values, initialValues)) {
        onValueChange(values[id]);
      }
    },
  });

  useEffect(() => {
    resetForm();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [operator]);

  const handleValueChange = ({ imaginaryTarget: { value: newValue, id: inputId } }) => {
    setFieldValue(inputId, +newValue);
  };

  const handleInputBlur = e => {
    handleBlur(e);
    submitForm();
  };

  if (isReadOnly) {
    return (
      <span className={getPrefixedClasses(['--text-space-left', '--rule-value-readonly'])}>
        {values[id]}
      </span>
    );
  }

  return (
    <NumberInput
      allowEmpty
      data-testid="date-rule-period-equal"
      id={id}
      invalid={!!errors[id]}
      invalidText={errors[id]}
      label=""
      name={id}
      onBlur={handleInputBlur}
      onChange={handleValueChange}
      placeholder={t('Enter value')}
      value={value}
    />
  );
};

PeriodEqual.propTypes = {
  id: PropTypes.string.isRequired,
  onValueChange: PropTypes.func.isRequired,
  operator: PROPTYPES.OPERATOR.isRequired,
  value: PropTypes.number,
};

PeriodEqual.defaultProps = {
  value: undefined,
};

export default PeriodEqual;
