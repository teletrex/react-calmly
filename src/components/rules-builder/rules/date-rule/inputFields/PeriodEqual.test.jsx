/* <LICENSE>
* 
* Copyright (C) 2022 Louis F. Roehrs, All rights reserved.
* 
* </LICENSE>
*  */




import React from 'react';
import { fireEvent, render, waitFor } from '@testing-library/react';

import { PERIOD_EQUAL_OPERATORS } from '../../constants/operators';
import { renderWithContext } from '../../../utils/testHelpers';

import PeriodEqual from './PeriodEqual';

const initialValue = 7;
const initialOperator = PERIOD_EQUAL_OPERATORS[2];

const props = {
  id: 'TestDateRuleElement',
  operator: initialOperator,
  onValueChange: jest.fn(),
};

describe('PeriodEqual', () => {
  beforeEach(() => jest.resetAllMocks());

  it('should render correctly', () => {
    const { container } = render(<PeriodEqual {...props} value={initialValue} />);
    expect(container).toMatchSnapshot();
  });

  it('should render correctly with a month operator', () => {
    const { container } = render(
      <PeriodEqual {...props} operator={PERIOD_EQUAL_OPERATORS[1]} value={1} />
    );
    expect(container).toMatchSnapshot();
  });

  it('should render correctly with a year operator', () => {
    const { container } = render(
      <PeriodEqual {...props} operator={PERIOD_EQUAL_OPERATORS[0]} value={2019} />
    );
    expect(container).toMatchSnapshot();
  });

  it('should render correctly for read-only', () => {
    const { container } = renderWithContext(<PeriodEqual {...props} value={initialValue} />, {
      isReadOnly: true,
    });
    expect(container).toMatchSnapshot();
  });

  it('should set the rule value when a value is changed and input field is blurred', async () => {
    const { getByPlaceholderText } = render(<PeriodEqual {...props} value={initialValue} />);
    const numberInput = getByPlaceholderText('Enter value');
    const newValue = 5;

    fireEvent.change(numberInput, { target: { value: newValue } });
    await waitFor(() => expect(props.onValueChange).toHaveBeenCalledTimes(0));

    fireEvent.blur(numberInput);
    await waitFor(() => expect(props.onValueChange).toHaveBeenCalledWith(newValue));
  });

  it('should reset validation on operator change', async () => {
    const { findByText, getByPlaceholderText, rerender } = render(
      <PeriodEqual {...props} value={123456789} />
    );

    fireEvent.blur(getByPlaceholderText('Enter value')); // trigger validation
    const validationMessage = await findByText('Value must be less than or equal to', {
      exact: false,
    });

    expect(validationMessage).toBeInTheDocument();

    rerender(<PeriodEqual {...props} operator={PERIOD_EQUAL_OPERATORS[0]} />);
    expect(validationMessage).not.toBeInTheDocument();
  });

  it('should not update the rule if the value was not changed', async () => {
    const { getByPlaceholderText } = render(<PeriodEqual {...props} value={initialValue} />);
    const numberInput = getByPlaceholderText('Enter value');

    fireEvent.blur(numberInput);
    await waitFor(() => expect(props.onValueChange).toHaveBeenCalledTimes(0));
  });
});
