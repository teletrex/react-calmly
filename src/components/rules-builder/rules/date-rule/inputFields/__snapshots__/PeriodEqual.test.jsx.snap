// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PeriodEqual should render correctly 1`] = `
<div>
  <div
    class="cds--number-input-wrapper"
  >
    <div
      class="cds--form-item"
    >
      <div
        class="cds--number cds--number--helpertext cds--number-input-wrapper_carbon-input"
      >
        <div
          class="cds--number__input-wrapper"
        >
          <input
            aria-label="Numeric input field with increment and decrement buttons"
            data-testid="date-rule-period-equal"
            id="TestDateRuleElement"
            max="Infinity"
            min="-Infinity"
            name="TestDateRuleElement"
            pattern="[0-9]*"
            placeholder="Enter value"
            step="1"
            type="number"
            value="7"
          />
          <div
            class="cds--number__controls"
          >
            <button
              aria-atomic="true"
              aria-label="Increment number. Press Space or Enter in order to use icon."
              aria-live="polite"
              class="cds--number__control-btn up-icon"
              focusable="false"
              tabindex="-1"
              title="Increment number. Press Space or Enter in order to use icon."
              type="button"
              unselectable="on"
            >
              <svg
                aria-hidden="true"
                class="up-icon"
                focusable="false"
                height="4"
                preserveAspectRatio="xMidYMid meet"
                style="will-change: transform;"
                viewBox="0 0 8 4"
                width="8"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 4l4-4 4 4z"
                />
              </svg>
            </button>
            <button
              aria-atomic="true"
              aria-label="Decrement number. Press Space or Enter in order to use icon."
              aria-live="polite"
              class="cds--number__control-btn down-icon"
              focusable="false"
              tabindex="-1"
              title="Decrement number. Press Space or Enter in order to use icon."
              type="button"
              unselectable="on"
            >
              <svg
                aria-hidden="true"
                class="down-icon"
                focusable="false"
                height="4"
                preserveAspectRatio="xMidYMid meet"
                style="will-change: transform;"
                viewBox="0 0 8 4"
                width="8"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 0L4 4 0 0z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>

  </div>
</div>
`;

exports[`PeriodEqual should render correctly for read-only 1`] = `
<div>
  <span
    class="cds--text-space-left cds--rule-value-readonly"
  >
    7
  </span>
</div>
`;

exports[`PeriodEqual should render correctly with a month operator 1`] = `
<div>
  <div
    class="cds--number-input-wrapper"
  >
    <div
      class="cds--form-item"
    >
      <div
        class="cds--number cds--number--helpertext cds--number-input-wrapper_carbon-input"
      >
        <div
          class="cds--number__input-wrapper"
        >
          <input
            aria-label="Numeric input field with increment and decrement buttons"
            data-testid="date-rule-period-equal"
            id="TestDateRuleElement"
            max="Infinity"
            min="-Infinity"
            name="TestDateRuleElement"
            pattern="[0-9]*"
            placeholder="Enter value"
            step="1"
            type="number"
            value="1"
          />
          <div
            class="cds--number__controls"
          >
            <button
              aria-atomic="true"
              aria-label="Increment number. Press Space or Enter in order to use icon."
              aria-live="polite"
              class="cds--number__control-btn up-icon"
              focusable="false"
              tabindex="-1"
              title="Increment number. Press Space or Enter in order to use icon."
              type="button"
              unselectable="on"
            >
              <svg
                aria-hidden="true"
                class="up-icon"
                focusable="false"
                height="4"
                preserveAspectRatio="xMidYMid meet"
                style="will-change: transform;"
                viewBox="0 0 8 4"
                width="8"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 4l4-4 4 4z"
                />
              </svg>
            </button>
            <button
              aria-atomic="true"
              aria-label="Decrement number. Press Space or Enter in order to use icon."
              aria-live="polite"
              class="cds--number__control-btn down-icon"
              focusable="false"
              tabindex="-1"
              title="Decrement number. Press Space or Enter in order to use icon."
              type="button"
              unselectable="on"
            >
              <svg
                aria-hidden="true"
                class="down-icon"
                focusable="false"
                height="4"
                preserveAspectRatio="xMidYMid meet"
                style="will-change: transform;"
                viewBox="0 0 8 4"
                width="8"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 0L4 4 0 0z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>

  </div>
</div>
`;

exports[`PeriodEqual should render correctly with a year operator 1`] = `
<div>
  <div
    class="cds--number-input-wrapper"
  >
    <div
      class="cds--form-item"
    >
      <div
        class="cds--number cds--number--helpertext cds--number-input-wrapper_carbon-input"
      >
        <div
          class="cds--number__input-wrapper"
        >
          <input
            aria-label="Numeric input field with increment and decrement buttons"
            data-testid="date-rule-period-equal"
            id="TestDateRuleElement"
            max="Infinity"
            min="-Infinity"
            name="TestDateRuleElement"
            pattern="[0-9]*"
            placeholder="Enter value"
            step="1"
            type="number"
            value="2019"
          />
          <div
            class="cds--number__controls"
          >
            <button
              aria-atomic="true"
              aria-label="Increment number. Press Space or Enter in order to use icon."
              aria-live="polite"
              class="cds--number__control-btn up-icon"
              focusable="false"
              tabindex="-1"
              title="Increment number. Press Space or Enter in order to use icon."
              type="button"
              unselectable="on"
            >
              <svg
                aria-hidden="true"
                class="up-icon"
                focusable="false"
                height="4"
                preserveAspectRatio="xMidYMid meet"
                style="will-change: transform;"
                viewBox="0 0 8 4"
                width="8"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 4l4-4 4 4z"
                />
              </svg>
            </button>
            <button
              aria-atomic="true"
              aria-label="Decrement number. Press Space or Enter in order to use icon."
              aria-live="polite"
              class="cds--number__control-btn down-icon"
              focusable="false"
              tabindex="-1"
              title="Decrement number. Press Space or Enter in order to use icon."
              type="button"
              unselectable="on"
            >
              <svg
                aria-hidden="true"
                class="down-icon"
                focusable="false"
                height="4"
                preserveAspectRatio="xMidYMid meet"
                style="will-change: transform;"
                viewBox="0 0 8 4"
                width="8"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 0L4 4 0 0z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>

  </div>
</div>
`;
